res/qml/Deck.qml

```diff
@@ -35,380 +161,708 @@ Item {
+                            break;
+                        case Deck.Cardinality.East:
+                            candidate.completed = sourceCenter.x - drag.source.width/2 < currentCenter.x
+                            break;
+                        case Deck.Cardinality.South:
+                            candidate.completed = drag.source.y < currentCenter.y
+                            break;
+                    }
+
+                    if (!candidate.completed) {
+                        return
+                    }
+
+                    let delta = Qt.point(drag.source.beginDrag.x - target.x, drag.source.beginDrag.y - target.y)
+                    let currentParent = drag.source.parent;
+                    let reverse = []
+                    while (currentParent != topParent) {
+                        reverse.push(currentParent.index)
+                        currentParent = currentParent.parent
+                    }
+                    let modelRef = root.minimized ? minimizedItemModel : itemModel
+                    while (reverse.length > 0) {
+                        modelRef = modelRef.get(reverse.pop()).items
+                    }
+                    let spacing
+
+                    switch (candidate.cardinality) {
+                        case Deck.Cardinality.West: {
+                            spacing = target.x - drag.source.beginDrag.x - drag.source.width
+                            target.x -= drag.source.width + spacing
+                            drag.source.beginDrag.x += target.width + spacing
+
+                            if (drag.source.move) {
+                                drag.source.move.source++
+                                } else {
+                                drag.source.move = {
+                                    ref: modelRef,
+                                    source: i,
+                                    target: sourceIdx,
+                                    count: 1
+                                }
+                            }
+                        }
+                        break;
+                        case Deck.Cardinality.East: {
+                            spacing = drag.source.beginDrag.x - target.x - target.width
+                            target.x += drag.source.width + spacing
+                            drag.source.beginDrag.x -= target.width + spacing
+
+                            if (drag.source.move) {
+                                drag.source.move.target = i
+                                drag.source.move.count++
+                                } else {
+                                drag.source.move = {
+                                    ref: modelRef,
+                                    source: sourceIdx,
+                                    target: i,
+                                    count: 1
+                                }
+                            }
```

Comment:
lots of duplicate code here, can we simplify?

Reply:
I tried to refactor that into sub functions, hopefully looks simpler!